### ISSUE TITLE
Chaining multiple WHENs in one scope

### ISSUE THREAD

**Issue by KKoovalsky** [philsquared/Catch#1067]
To chain multiple WHENs I have to do it in such a way:
```
GIVEN("Some background") {
    WHEN("Something happens") {
        THEN("The result should be") {
            AND_WHEN("Other things happen") {
                ....
```
When I want to chain multiple WHENs the code could grow, what will lead to ugliness of scenario.  
I want to accomplish the same behaviour, but with such syntax:
```
GIVEN("Some background") {
    WHEN("Something happens") { // When no. 1
        THEN("The result should be") {
            ....
        }
    AND_WHEN("Other things happen after "the something" ") {  // When no. 2
        THEN("The result should be") {
            ...
```
I don't want the "When no. 1" and the "When no. 2" to be executed in different test runs.  
Is there any workaround to accomplish this?

**Comment by alwaysmpe** [comment]
I've been likewise looking at this today due to slowing tests, came here to raise a similar query. the conclusion I've reached so far is that it is possible to refactor each WHEN block into different functions (my purpose was to validate each stage of an iterative process so a recursive function) but that is undesirable. Each when/then statement shows up on the command line if any statements fail but does behave as desired. It's likewise possible to always perform each step and only validate the results, i.e. have "When no. 1" and "When no. 2" outside of the when/and_when block and only have the when/then/require sections run once but if, as is for my purposes, when 1, when 2...N are slow operations repeating them N times is not a feasible solution.

**Comment by KKoovalsky** [comment]
Yeah, I put this single steps into TEST_CASE and they are executed sequentially (according to the example in my previous post):
```
// Given some background
give_some_background();

// When something happens
REQUIRE_NO_THROW(do_something());

// Then the result should be
REQUIRE(get_some_val() == 15);

// And when other things happen
...
```
I don't care about slow code execution because I test time dependent operations. My only goal is to provide readability.

Another workaround could be to deceive auto-formatter (like clang-format) in some way:
```
GIVEN("Some background") {
    WHEN("Something happens") {
        THEN("The result should be") {
			...
    AND_WHEN("Other things happen") {
        THEN("The result will change") {
			...
	AND_WHEN("The app's stack overflows") {
		THEN("There will be undefined behaviour") {
			...
		}
	}}}}}
}
```

**Comment by kennymalac** [comment]
This is really painful to deal with!
